hail/src/main/scala/is/hail/annotations/RegionValueBuilder.scala

```diff
@@ -167,6 +167,33 @@ class RegionValueBuilder(var region: Region) {
     val length = t.loadLength(region, aoff)
     assert(length == indexstk.top)
 
+    endArrayUnchecked()
+  }
+
+  // using this function, rather than startArray will set all elements of the array to missing by
+  // default, you will need to use setPresent to add a value to this array.
+  def startMissingArray(length: Int, init: Boolean = true) {
+    val t = currentType().asInstanceOf[PArray]
+    if (t.elementType.required)
```

Comment:
This looks like it's exactly the same code as `startArray`. Is there a way to refactor these to utilize common code. Maybe a flag on `startArray(missing=true)`?